framework_tvb/tvb/core/data_encryption_handler.py

```diff
@@ -0,0 +1,193 @@
+# -*- coding: utf-8 -*-
+#
+#
+# TheVirtualBrain-Framework Package. This package holds all Data Management, and 
+# Web-UI helpful to run brain-simulations. To use it, you also need do download
+# TheVirtualBrain-Scientific Package (for simulators). See content of the
+# documentation-folder for more details. See also http://www.thevirtualbrain.org
+#
+# (c) 2012-2020, Baycrest Centre for Geriatric Care ("Baycrest") and others
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software Foundation,
+# either version 3 of the License, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License along with this
+# program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+#   CITATION:
+# When using The Virtual Brain for scientific publications, please cite it as follows:
+#
+#   Paula Sanz Leon, Stuart A. Knock, M. Marmaduke Woodman, Lia Domide,
+#   Jochen Mersmann, Anthony R. McIntosh, Viktor Jirsa (2013)
+#       The Virtual Brain: a simulator of primate brain network dynamics.
+#   Frontiers in Neuroinformatics (7:10. doi: 10.3389/fninf.2013.00010)
+#
+#
+"""
+.. moduleauthor:: Bogdan Valean <bogdan.valean@codemart.ro>
+"""
+
+import os
+import shutil
+import threading
+from queue import Queue
+from threading import Lock
+
+from syncrypto import Crypto, Syncrypto
+from tvb.basic.logger.builder import get_logger
+from tvb.basic.profile import TvbProfile
+from tvb.core.decorators import synchronized
+
+LOGGER = get_logger(__name__)
+
+# Queue used to push projects which need synchronisation
+sync_project_queue = Queue()
+
+# Dict used to count the project frequency from the queue
+queue_elements_count = {}
+
+# Dict used to count how the project usage
+users_project_usage = {}
+
+# Dict used to keep projects which are marked for deletion
+marked_for_delete = set()
+
+lock = Lock()
```

Comment:
these seem like the properties of a singleton.
while staying on the module does effectively the same job, from an oop architecture pov woudn't be nicer to have an actual class ?